api/utxoCache.ts

```diff
@@ -0,0 +1,109 @@
+import { getAddressUtxoOrdinalBundles, getUtxoOrdinalBundle } from './ordinals';
+
+export type UtxoStates = 'inscribed' | 'notInscribed' | 'unknown';
+
+export type UtxoCacheStruct = {
+  [utxoId: string]: UtxoStates;
+};
+
+export type StorageAdapter = {
+  get(key: string): Promise<string | null> | string | null;
+  set(key: string, value: string): Promise<void> | void;
+  remove(key: string): Promise<void> | void;
+};
+
+type UtxoCacheConfig = {
+  cacheStorageController: StorageAdapter;
+};
+
+export class UtxoCache {
+  private readonly _cacheStorageController: StorageAdapter;
+
+  CACHE_STORAGE_KEY = 'utxoCache';
+
+  CACHE_VERSION_STORAGE_KEY = 'utxoCacheVersion';
+
+  VERSION = 1;
+
+  constructor(config: UtxoCacheConfig) {
+    this._cacheStorageController = config.cacheStorageController;
+  }
+
+  _getCache = async (): Promise<UtxoCacheStruct> => {
```

Comment:
Could you make these private? 🙏 The underscore is not super necessary if they are private, but I guess we should decide on that naming convention.
```suggestion
  private _getCache = async (): Promise<UtxoCacheStruct> => {
```